pkg/v1/tkg/utils/common.go

```diff
@@ -295,3 +298,83 @@ func CompareMajorMinorPatchVersion(version1, version2 string) bool {
 	}
 	return false
 }
+
+// Getter interface defines methods that a Cluster API object should implement in order to
+// use the conditions package for getting conditions.
+type Getter interface {
+	crtclient.Object
+	GetConditions() clusterv1alpha3.Conditions
+}
+
+// Get returns the condition with the given type, if the condition does not exists,
+// it returns nil.
+func Get(from Getter, t clusterv1alpha3.ConditionType) *clusterv1alpha3.Condition {
+	conditions := from.GetConditions()
+	if conditions == nil {
+		return nil
+	}
+
+	for _, condition := range conditions {
+		if condition.Type == t {
+			return &condition
+		}
+	}
+	return nil
+}
+
+// Has returns true if a condition with the given type exists.
+func Has(from Getter, t clusterv1alpha3.ConditionType) bool {
+	return Get(from, t) != nil
+}
+
+// IsTrue is true if the condition with the given type is True, otherwise it return false
+// if the condition is not True or if the condition does not exist (is nil).
+func IsTrue(from Getter, t clusterv1alpha3.ConditionType) bool {
+	if c := Get(from, t); c != nil {
+		return c.Status == corev1.ConditionTrue
+	}
+	return false
+}
+
+// IsFalse is true if the condition with the given type is False, otherwise it return false
+// if the condition is not False or if the condition does not exist (is nil).
+func IsFalse(from Getter, t clusterv1alpha3.ConditionType) bool {
+	if c := Get(from, t); c != nil {
+		return c.Status == corev1.ConditionFalse
+	}
+	return false
+}
+
+// IsUnknown is true if the condition with the given type is Unknown or if the condition
+// does not exist (is nil).
+func IsUnknown(from Getter, t clusterv1alpha3.ConditionType) bool {
+	if c := Get(from, t); c != nil {
+		return c.Status == corev1.ConditionUnknown
+	}
+	return true
+}
+
+// GetReason returns a nil safe string of Reason for the condition with the given type.
+func GetReason(from Getter, t clusterv1alpha3.ConditionType) string {
+	if c := Get(from, t); c != nil {
+		return c.Reason
+	}
+	return ""
+}
+
+// GetMessage returns a nil safe string of Message.
+func GetMessage(from Getter, t clusterv1alpha3.ConditionType) string {
+	if c := Get(from, t); c != nil {
+		return c.Message
+	}
+	return ""
+}
+
+// GetSeverity returns the condition Severity or nil if the condition
+// does not exist (is nil).
+func GetSeverity(from Getter, t clusterv1alpha3.ConditionType) *clusterv1alpha3.ConditionSeverity {
+	if c := Get(from, t); c != nil {
+		return &c.Severity
+	}
+	return nil
+}
```

Comment:
Are we duplicating this from ClusterAPI? Can we use the original implementation instead?

Reply:
This is one of the consequences of the CLI to support both TKGm(CAPI v1beta1) and TKGS(v1alpha3). TKGS recently move to v1alpha2 which has support for CAPI conditions. In order to  leverage the conditions API and to support both TKGm and TKGS, we had no other option other than to copy some of the conditions API functionality. Hope we won't have this problem when we move to Unified TKG.

Reply:
If there's no way around it, can we put the transplanted code into a separate file, at least? Also, add a TODO comment (at the top of the file) to remove this code, specifying conditions that would allow us to do so.

Reply:
Agreed. Seperated those utils into a seperate file and added a TODO comment.